pinot-core/src/main/java/com/linkedin/pinot/core/io/util/PinotDataBitSet.java

```diff
@@ -39,6 +39,20 @@
     }
   }
 
+  public static int getNumBitsPerValue(int cardinality) {
+    // Use at least one bit even there is only one possible value
```

Comment:
There is no **Math.log2()**. In order to do so, we have to do **Math.ceil(Math.log(cardinality) / Math.log(2))**, while I am afraid that using double division could cause wrong result because of the precision error.
As for the **Math.max()**, we can avoid the actual calculation by using the if condition check.